src/CLaSH/Prelude/BlockRam.hs

```diff
@@ -528,3 +536,22 @@ blockRam# clk content wr rd en din = register' clk undefined dout
       d' <- readArray ram r
       when e (writeArray ram w d)
       return d'
+
+-- | Create read-after-write blockRAM from a read-before-write one (synchronised to specified clock)
+--
+readNew' :: Eq addr => SClock clk -> (Signal' clk addr -> Signal' clk addr -> Signal' clk Bool -> Signal' clk a -> Signal' clk a) -> Signal' clk addr -> Signal' clk addr -> Signal' clk Bool -> Signal' clk a -> Signal' clk a
+readNew' clk ram wrAddr rdAddr wrEn wrData = mux wasSame wasWritten $ ram wrAddr rdAddr wrEn wrData
+  where sameAddr = (==) <$> wrAddr <*> rdAddr
+        wasSame = register' clk False ((&&) <$> wrEn <*> sameAddr)
+        wasWritten = register' clk undefined wrData
+        mux = liftA3 (\b t f -> if b then t else f)
```

Comment:
`mux` already exists in [`CLaSH.Signal`](https://github.com/clash-lang/clash-prelude/blob/master/src/CLaSH/Signal.hs#L20).